storage/series/buffer.go

```diff
@@ -581,9 +580,12 @@ func (b *dbBufferBucket) streams(ctx context.Context) []xio.SegmentReader {
 		}
 	}
 	for i := range b.encoders {
+		start := b.start
+		end := b.lastRead()
```

Comment:
Can't use last read here either, needs to use `end := start.Add(b.opts.RetentionOptions().BlockSize())` or similar.